### PR TITLE
Check for GPG in configure, and other build tweaks

### PR DESCRIPTION
**Pull request by wjt**
https://phabricator.endlessm.com/T13661

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -139,6 +139,13 @@ fi
 GLIB_TESTS
 LIBGLNX_CONFIGURE
 
+# Both the tests and the application invoke the GPG executable
+AC_ARG_VAR([GPG_PATH], [GPG executable])
+AC_PATH_PROG([GPG_PATH], [gpg])
+AS_IF([test -z "$GPG_PATH"], [AC_MSG_ERROR([GPG not found])])
+AC_SUBST([GPG_PATH])
```
Comment (pwithnall) [inline configure.ac]: Nitpick: The [`AC_ARG_VAR` documentation](https://www.gnu.org/software/autoconf/manual/autoconf-2.64/html_node/Setting-Output-Variables.html#index-AC_005fARG_005fVAR-1030) says that it implicitly does an `AC_SUBST` on the variable, so you don’t strictly need this `AC_SUBST` (but it does no harm).

**Review by pwithnall — APPROVED** [review]
LGTM with a nitpick which you could or could not fix before pushing.

File: configure.ac (hunk 1 above)
Comment (wjt) [reply configure.ac]: Good spot; I read this too but I think I got confused between `AC_SUBST` and `AC_DEFINE`.

**Comment by pwithnall** [comment]
> I got confused between `AC_SUBST` and `AC_DEFINE`.

Don’t we all? :-(

**Pull request merged**